crates/neon-runtime/src/napi/call.rs

```diff
@@ -17,19 +19,79 @@ impl Default for CCallback {
     }
 }
 
-pub unsafe extern "C" fn set_return(_info: &FunctionCallbackInfo, _value: Local) { unimplemented!() }
+pub unsafe extern "C" fn set_return(_info: FunctionCallbackInfo, _value: Local) {
+
+}
 
-pub unsafe extern "C" fn get_isolate(_info: &FunctionCallbackInfo) -> Env { unimplemented!() }
+pub unsafe extern "C" fn get_isolate(info: FunctionCallbackInfo) -> Env { unimplemented!() }
 
 // FIXME: Remove. This will never be implemented
 pub unsafe extern "C" fn current_isolate() -> Env { panic!("current_isolate won't be implemented in n-api") }
 
-pub unsafe extern "C" fn is_construct(_info: &FunctionCallbackInfo) -> bool { unimplemented!() }
+pub unsafe extern "C" fn is_construct(_info: FunctionCallbackInfo) -> bool { unimplemented!() }
 
-pub unsafe extern "C" fn this(_info: &FunctionCallbackInfo, _out: &mut Local) { unimplemented!() }
+pub unsafe extern "C" fn this(env: Env, info: FunctionCallbackInfo, out: &mut Local) {
+    let status = napi::napi_get_cb_info(
+        env,
+        info,
+        null_mut(),
+        null_mut(),
+        out as *mut _,
+        null_mut(),
+    );
+    assert_eq!(status, napi::napi_status::napi_ok);
+}
 
-pub unsafe extern "C" fn data(_info: &FunctionCallbackInfo, _out: &mut Local) { unimplemented!() }
+/// Mutates the `out` argument provided to refer to the associated data value of the
+/// `napi_callback_info`.
+pub unsafe extern "C" fn data(env: Env, info: FunctionCallbackInfo, out: &mut *mut c_void) {
+    let mut data = null_mut();
+    let mut argc = 0usize;
```

Comment:
This particular use is now removed, but the same `0usize` occurs later inside `len()`. There, it's inferred as i32 because the value is returned as `argc as i32`, so we have to tell Rust that it's an usize somehow. The other option would be to do
```rust
let mut argc: usize = 0;
```
but i feel like that makes the binding name kinda hard to see?

Reply:
I'll go with your instincts. :)